subprojects/docs/src/docs/userguide/upgrade_version_4.adoc

```diff
@@ -0,0 +1,493 @@
+ * `getDestinationDir()` was replaced by `getDestinationDirectory()`.
+ * `getBinaryFile()`, `getOutputFile()` was replaced by `getLinkedFile()`.
+ * `setOutputFile(File)` was removed. Use `Property.set()` instead.
+ * `setOutputFile(Provider)` was removed. Use `Property.set()` instead.
+ * `getTargetPlatform()` was changed to return a `Property`.
+ * `setTargetPlatform(NativePlatform)` was removed. Use `Property.set()` instead.
+ * `getToolChain()` was changed to return a `Property`.
+ * `setToolChain(NativeToolChain)` was removed. Use `Property.set()` instead.
+
+link:{groovyDslPath}/org.gradle.nativeplatform.tasks.CreateStaticLibrary.html[CreateStaticLibrary]::
+ * `getOutputFile()` was changed to return a `Property`.
+ * `setOutputFile(File)` was removed. Use `Property.set()` instead.
+ * `setOutputFile(Provider)` was removed. Use `Property.set()` instead.
+ * `getTargetPlatform()` was changed to return a `Property`.
+ * `setTargetPlatform(NativePlatform)` was removed. Use `Property.set()` instead.
+ * `getToolChain()` was changed to return a `Property`.
+ * `setToolChain(NativeToolChain)` was removed. Use `Property.set()` instead.
+ * `getStaticLibArgs()` was changed to return a `ListProperty`.
+ * `setStaticLibArgs(List)` was removed. Use `ListProperty.set()` instead.
+
+link:{groovyDslPath}/org.gradle.nativeplatform.tasks.InstallExecutable.html[InstallExecutable]::
+ * `getSourceFile()` was replaced by `getExecutableFile()`.
+ * `getPlatform()` was replaced by `getTargetPlatform()`.
+ * `setTargetPlatform(NativePlatform)` was removed. Use `Property.set()` instead.
+ * `getToolChain()` was changed to return a `Property`.
+ * `setToolChain(NativeToolChain)` was removed. Use `Property.set()` instead.
+
+The following have also seen similar changes:
+
+ * link:{groovyDslPath}/org.gradle.language.assembler.tasks.Assemble.html[Assemble]
+ * link:{groovyDslPath}/org.gradle.language.rc.tasks.WindowsResourceCompile.html[WindowsResourceCompile]
+ * link:{javadocPath}/org/gradle/nativeplatform/tasks/StripSymbols.html[StripSymbols]
+ * link:{javadocPath}/org/gradle/nativeplatform/tasks/ExtractSymbols.html[ExtractSymbols]
+ * link:{javadocPath}/org/gradle/language/swift/tasks/SwiftCompile.html[SwiftCompile]
+ * link:{javadocPath}/org/gradle/nativeplatform/tasks/LinkMachOBundle.html[LinkMachOBundle]
+
+[[rel4.5:bom_import]]
+=== [4.5] BOM import
+
+Gradle now provides support for importing bill of materials (BOM) files, which are effectively POM files that use `<dependencyManagement>` sections to control the versions of direct and transitive dependencies. All you need to do is declare the POM as just another dependency.
+
+The following example picks the versions of the `gson` and `dom4j` dependencies from the declared Spring Boot BOM:
+
+----
+dependencies {
+    // import a BOM
+    implementation 'org.springframework.boot:spring-boot-dependencies:1.5.8.RELEASE'
```

Comment:
I think the @gradle/dependency-management wanted to change how this works for 5.0, so they'll want to be aware of this document.

Reply:
I think most teams will need to review this before it goes live. BTW, if that team does pop in here, the review should be happening at #6165. Sorry for the confusion.